src/sweeps/bayes_search.py

```diff
@@ -407,19 +407,23 @@ def _construct_gp_data(
 
     X_norms = params.normalize_runs_as_array(runs)
     worst_metric = impute(goal, metric_name, ImputeStrategy.worst, runs=runs)
+
+    def get_metric():
+        try:
+            return run.metric_extremum(
+                metric_name, kind="maximum" if goal == "maximize" else "minimum"
+            )
+        except ValueError:
+            if impute_strategy != "worst":
+                return impute(
+                    goal, metric_name, impute_strategy, run=run, runs=runs
+                )  # default
+            else:
+                return worst_metric
+
     for run, X_norm in zip(runs, X_norms):
         if run.state == RunState.finished:
-            try:
-                metric = run.metric_extremum(
-                    metric_name, kind="maximum" if goal == "maximize" else "minimum"
-                )
-            except ValueError:
-                if impute_strategy != "worst":
-                    metric = impute(
-                        goal, metric_name, impute_strategy, run=run, runs=runs
-                    )  # default
-                else:
-                    metric = worst_metric
+            metric = get_metric()
```

Comment:
we should separate concerns and keep the impute strategy for failed runs separate from the impute strategy for running runs. it might be that someone wants to impute failed runs with the worst value of the metric and running runs with the best. we should allow that and use the appropriate config setting in each of the calls to `get_metric()` here